File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/shipping/services/carrier_gateway.py
"""CarrierGateway for the shipping domain."""

from typing import Optional

from metaengine_demo.shipping.aggregates.shipment import Shipment


class CarrierGateway:
    """Gateway service that delegates shipment operations to carriers."""

    def create(self, input: Optional[Shipment]) -> Shipment:
        """Hand off a shipment to a carrier."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Shipment]:
        """Query a carrier for shipment details by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Shipment]:
        """List carrier-known shipments up to the given limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Cancel a shipment with the carrier by id."""
        raise NotImplementedError("not implemented")

File: benchmark/results/java-invocation-script/run-001/b-baseline/output/com/metaengine/demo/shipping/services/CarrierGateway.java
package com.metaengine.demo.shipping.services;

import com.metaengine.demo.shipping.aggregates.Shipment;
import java.util.List;

/** CarrierGateway for the shipping domain. */
public class CarrierGateway {

    /** Hand off a Shipment to the carrier. */
    public Shipment create(Shipment input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Look up a Shipment by id; may return null. */
    public Shipment findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List shipments up to the given limit. */
    public List<Shipment> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a shipment by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}
